lib/tasks/valid_test_data_generator.rb

```diff
@@ -108,7 +108,37 @@ def weighted_choice(selection:, odds:)
   class AmbitionSpecificPopulater < LeadProviderPopulater
     class << self
       def call(name:, total_schools: 3, participants_per_school: 3000)
-        new(name: name).call(total_schools: total_schools, participants_per_school: participants_per_school)
+        generator = new(name: name)
+        generator.remove_old_data(created_at: "2021-08-18 13:43".."2021-08-18 13:49")
+        generator.call(total_schools: total_schools, participants_per_school: participants_per_school)
+      end
+    end
+
+    def remove_old_data(created_at:)
+      participants = lead_provider.ecf_participants.where(created_at: created_at)
+      participants.all.each do |participant|
+        tp = participant.teacher_profile
+        # Just in case the withdrawn merge happens before this is run.
+        # tp.participant_profiles.each do |pp|
+        #   pp.participant_profile_states.destroy_all!
+        # end
+        tp.destroy!
```

Comment:
I hope we don't need that, we have 
```
  has_one :teacher_profile, dependent: :destroy
```
on `user` model.

Reply:
I had thought about doing an .each(&:destroy!) on participants, but I thought this would make it more explicit and let me double check the ordering of creation/destruction if I had to do it again later.

Reply:
Since it's now in a commit, I'll make that change too.